src/main/java/org/jenkinsci/plugins/envinject/EnvInjectBuildWrapper.java

```diff
@@ -104,8 +101,8 @@ public void buildEnvVars(Map<String, String> env) {
                     env.putAll(resultVariables);
                 }
             };
-        } catch (Throwable throwable) {
-            logger.error("Problems occurs on injecting env vars as a build wrap: " + throwable.getCause());
+        } catch (Exception e) {
```

Comment:
previous in case of OOM the build was set to FAILURE and there is no chance to change the state of build.

Typycally catch Throwable is bed practice and the OOM happens in the same JVM of jenkins so, now IT would be informed that jenkins instance should be restarted or some JVM setting be should be adjust to avoid again issue.
Previously an OOM (for example) was eated and an incomprensible message was printed (null :\) so anyone would be alern of the problem until happen in other point.